kuka_agilus_support/urdf/kr6_r700_sixx_macro.xacro

```diff
@@ -2,110 +2,120 @@
 <robot xmlns:xacro="http://wiki.ros.org/xacro">
   <xacro:include filename="$(find kuka_resources)/urdf/common_constants.xacro"/>
   <xacro:include filename="$(find kuka_resources)/urdf/common_materials.xacro"/>
+  <xacro:include filename="$(find kuka_agilus_support)/urdf/kr_agilus_ros2_control.xacro" />
+  <xacro:include filename="$(find kuka_agilus_support)/urdf/kr_agilus_transmission.xacro" />
+
+  <xacro:macro name="kuka_kr6_r700_sixx_robot" params="prefix client_ip client_port use_fake_hardware">
+    <!-- ros2 control instance -->
+    <xacro:kuka_agilus_ros2_control
+    name="KR6R700sixx" 
+    prefix="${prefix}"
+    client_ip="${client_ip}"
+    client_port="${client_port}"
+    use_fake_hardware="${use_fake_hardware}" />
 
-  <xacro:macro name="kuka_kr6r700sixx" params="prefix">
     <link name="${prefix}base_link">
       <visual>
         <origin xyz="0 0 0" rpy="0 0 0" />
         <geometry>
-          <mesh filename="package://kuka_kr6_support/meshes/kr6_agilus/visual/base_link.dae" />
+          <mesh filename="package://kuka_agilus_support/meshes/KR6R700sixx/visual/base_link.dae" />
```

Comment:
Quick comment here: it's a best/common practice in ROS to use lowercase characters for just about everything related to *resource names*. This includes things like TF frames, `link`s and `joint`s in URDFs and XACROs, but also file and directory names.

Auto-complete exists, but not many (text) editors will auto-complete elements from `package://` URIs (to give one example), and not having to "guess" the capitalisation of a resource makes things quite a bit easier then.

(I realise the type/model name does include capitalisation, but the original files converted all of that into a form which doesn't go against the `no-capitals-in-resource-names` convention)

For some other robot OEMs, we (as in: ROS-Industrial) have a codified procedure (ie: how to transform type/model names into ROS compatible names). Perhaps that would be something to define for KUKA as well?

Reply:
Thanks for your input, we will come up with a solution for model -> ros2_compatible naming and update the urdfs accordingly 😊

Reply:
Perhaps interesting in this context: https://github.com/ros-industrial/rep/blob/467b9f98ffb8207acc5f2810728197433f1213ef/rep-I0007.rst